### PR TITLE
fix: cloning app with seq cd pipeline fix

### LINKED ISSUE #2232: Bug: sequential pipelines get converted to parallel after cloning application
Create an application containing sequential pipelines.
After cloning the application, sequential pipelines get converted as parallel.

AB#165

### PR DESCRIPTION
**Pull request by prakash100198**
# Description

This pr aims to fix the conversion of sequential cd pipelines to parallel cd pipelines after cloning the application
Fixes #2232 

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] Tested locally as well as deploying the changes on cluster.
1. Making new app with multiple seq. and parallel cd, then cloning the app, 
2. Linked Ci and external Ci with seq. and parallel cds. 

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/appClone/AppCloneService.go [hunk 1]
```diff
@@ -594,24 +612,50 @@ func (impl *AppCloneServiceImpl) CreateWf(oldAppId, newAppId int, userId int32,
 			AppWorkflowMappingDto: nil, //first create new mapping then add it
 			UserId:                userId,
 		}
+		thisWf, err = impl.appWorkflowService.CreateAppWorkflow(thisWf)
+		if err != nil {
+			impl.logger.Errorw("error in creating workflow without external-ci", "err", err)
+			return nil, err
+		}
+		impl.logger.Debugw("app workflow created")
```
Comment (kripanshdevtron) [inline pkg/appClone/AppCloneService.go:620]: not needed

File: pkg/appClone/AppCloneService.go [hunk 2]
```diff
@@ -594,24 +612,50 @@ func (impl *AppCloneServiceImpl) CreateWf(oldAppId, newAppId int, userId int32,
...
+			externalCiPipelineId, err = impl.pipelineBuilder.CreateExternalCiAndAppWorkflowMapping(newAppId, thisWf.Id, userId, tx)
+			if err != nil {
+				impl.logger.Errorw("error in creating new external ci pipeline and new app workflow mapping", "refAppId", oldAppId, "newAppId", newAppId, "err", err)
+				return nil, err
+			}
+			err = tx.Commit()
 			if err != nil {
-				impl.logger.Errorw("errir in creating workflow without extenal-ci", "err", err)
 				return nil, err
 			}
```
Comment (kripanshdevtron) [inline pkg/appClone/AppCloneService.go:647]: can we take this logic to a func ??

Comment (prakash100198) [reply pkg/appClone/AppCloneService.go]: sure, have extracted it out in a func

File: pkg/appClone/AppCloneService.go (hunk 1 above)
Comment (prakash100198) [reply pkg/appClone/AppCloneService.go]: have removed this debug log, didn't need this really

**Pull request merged**